invenio_search_ui/assets/semantic-ui/js/invenio_search_ui/components/MultipleOptionSearchBar.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2022 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import { withState } from "react-searchkit";
+import { Search, Label } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_communities/i18next";
+import PropTypes from "prop-types";
+import ReactDOM from "react-dom";
+
+const resultRenderer = ({ text }, queryString) => {
+  return (
+    <div className="flex">
+      <div className="truncated pt-5">{queryString}</div>
+      <Label className="right-floated">{text}</Label>
+    </div>
+  );
+};
+
+export class MultipleOptionSearchBarWithInternalState extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      queryString: "",
+    };
+  }
+
+  render() {
+    const { placeholder, options } = this.props;
+    const { queryString } = this.state;
+
+    return (
+      <Search
+        fluid
+        onResultSelect={(e, { result }) => {
+          const { value: url } = result;
+          window.location = `${url}?q=${queryString}`;
+        }}
+        onSearchChange={(e, { value }) => {
+          this.setState({ queryString: value });
+        }}
+        resultRenderer={(props) => resultRenderer(props, queryString)}
+        results={options}
+        value={queryString}
+        placeholder={placeholder}
+        selectFirstResult
+      />
+    );
+  }
+}
+
+MultipleOptionSearchBarWithInternalState.propTypes = {
+  options: PropTypes.array.isRequired,
+  placeholder: PropTypes.string,
+};
+
+MultipleOptionSearchBarWithInternalState.defaultProps = {
+  placeholder: i18next.t("Search"),
+};
+
+export const addMultipleSearchBarToHeader = (
+  headerSearchbar,
+  placeholder,
+  ...options
+) => {
+  const searchBarOptions = JSON.parse(headerSearchbar.dataset.options);
```

Comment:
this is an unsafe access to `headerSearchbar.dataset.options`, I would consider using optional chaining and a default value for `searchBarOptions`. Worst case scenario an empty array is passed, instead of throwing an exception.

Reply:
I didn't want to accept an empty array because otherwise the search wouldn't work (the dropdown options are always required), that's the reason to force the options, if they are not set it should fail, however Nico is right, the parsing shouldn't happen in the cmp, thanks for that!